crates/red_knot_workspace/tests/check.rs

```diff
@@ -205,16 +205,25 @@ impl SourceOrderVisitor<'_> for PullTypesVisitor<'_> {
             }
             Stmt::Assign(assign) => {
                 for target in &assign.targets {
-                    self.visit_assign_target(target);
+                    self.visit_target(target);
                 }
+                // TODO
+                //self.visit_expr(&assign.value);
+                return;
+            }
+            Stmt::For(for_stmt) => {
+                self.visit_target(&for_stmt.target);
+                // TODO
+                //self.visit_expr(&for_stmt.iter);
```

Comment:
This is actually a bug, calling `visit_expr` on the expression panics in the following case:
```py
[] = *data
() = *data
```

This is because in `infer_target`, we completely skip the inference when there are no elements in the list / tuple target.

I'm going to fix this in a follow-up.